netbox/dcim/models/cables.py

```diff
@@ -529,46 +535,66 @@ def from_origin(cls, terminations):
                 object_to_path_node(t) for t in terminations
             ])
 
-            # Step 2: Determine the attached link (Cable or WirelessLink), if any
-            link = terminations[0].link
-            if link is None and len(path) == 1:
-                # If this is the start of the path and no link exists, return None
-                return None
-            elif link is None:
+            # Step 2: Determine the attached links (Cable or WirelessLink), if any
+            links = [termination.link for termination in terminations if termination.link is not None]
+            if len(links) == 0:
+                if len(path) == 1:
+                    # If this is the start of the path and no link exists, return None
+                    return None
                 # Otherwise, halt the trace if no link exists
                 break
-            assert type(link) in (Cable, WirelessLink)
+            assert all(type(link) in (Cable, WirelessLink) for link in links)
+
+            # Create set of links in path.  Cannot use list(set()) as it does this in a non-deterministic manner
+            links_path = []
+            for link in links:
+                if object_to_path_node(link) not in links_path:
+                    links_path.append(object_to_path_node(link))
 
-            # Step 3: Record the link and update path status if not "connected"
-            path.append([object_to_path_node(link)])
-            if hasattr(link, 'status') and link.status != LinkStatusChoices.STATUS_CONNECTED:
+            # Step 3: Record the links
+            path.append(links_path)
+
+            # Step 4: Update the path status if a link is not connected
+            links_status = [
+                link.status for link in links if hasattr(link, 'status') and
+                link.status != LinkStatusChoices.STATUS_CONNECTED
+            ]
+            if len(links_status) and len(links) != len(links_status):
```

Comment:
This can be simplified by using `any()` to check for any values other than `STATUS_CONNECTED`.